From 039f0ddc4426b1c48fe1d2da572a66b8c0dbc54b Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Thu, 12 Jan 2023 10:57:02 +0200
Subject: [PATCH] [SW-66341] gaudi2: razwi: remove dead code

gaudi2 preboot's is implemented in preboot-rtos,
while preboot project is used only for ppboot of
gaudi2. Hence, this patch remove a dead code,
which is supposed to run as part of gaudi2 preboot,
and as it is implemented in preboot-rtos project.

Change-Id: Ie566e8e8bce2277f4dd295e0d1c39149e0a3b2c3
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 7a8e1f0..8b924e2 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1175,30 +1175,6 @@
 	return boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK;
 }
 
-/*
- * TODO: This is a temporary function and
- * to be removed after GAUDI2 PO
- */
-void check_razwi_intr(void)
-{
-	u32 pattern = 0;
-	u32 reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
-	u32 bitmask = reg & PCIE_INTR_ERR_INTR_BIT;
-
-	if (bitmask & PCIE_HBW_ERR_INTR_BIT)
-		pattern = PCIE_HBW_INTR_STS_PATTERN;
-	else if (bitmask & PCIE_LBW_ERR_INTR_BIT)
-		pattern = PCIE_LBW_INTR_STS_PATTERN;
-	else if (bitmask & PCIE_BAD_ACC_ERR_INTR_BIT)
-		pattern = RAZWI_INTR_STATUS_PATTERN;
-
-	if (pattern) {
-		regmap_writel(pattern, mmGIC_RAZWI_STATUS_REG);
-		/* Reset the indication */
-		regmap_writel(reg & ~bitmask, mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
-	}
-}
-
 u32 dev_pci_dbi_cfg_offset_get(void)
 {
 	return (u32)mmPCIE_DBI_BASE & 0xFFFFFFF;
diff --git a/lib/common/cmd_module.c b/lib/common/cmd_module.c
index 586d2b9..3190f40 100644
--- a/lib/common/cmd_module.c
+++ b/lib/common/cmd_module.c
@@ -12,8 +12,6 @@
 #include "cmd_module.h"
 
 __weak bool preboot_host_gpio_asserted(void) { return true; }
-/* TODO: This is temporary and to be removed after GAUDI2 PO */
-__weak void check_razwi_intr(void) {}
 
 static void i2c_debug_mode_handling(void)
 {
@@ -43,9 +41,6 @@
 
 __weak void module_run_uncond_each_loop(void)
 {
-	/* TODO: This is temporary and to be removed after GAUDI2 PO */
-	check_razwi_intr();
-
 	i2c_debug_mode_handling();
 
 	cpu_boot_if_fix_ro_regs();
